core/runtime.rs

```diff
@@ -648,6 +648,38 @@ impl JsRuntime {
     scope.perform_microtask_checkpoint();
   }
 
+  /// Waits for the given value to resolve while polling the event loop.
+  ///
+  pub async fn resolve_value(
+    &mut self,
+    global: v8::Global<v8::Value>,
+  ) -> Result<v8::Global<v8::Value>, AnyError> {
+    poll_fn(|cx| {
+      self.poll_event_loop(cx, false);
```

Comment:
Suppose we should bail if the main loop poll yields `Poll::Ready` as nothing can progress?

Reply:
Yes